src/lib/boostagrams.js

```diff
@@ -0,0 +1,70 @@
+import fs from 'fs'
+import createDOMPurify from 'dompurify'
+import { JSDOM } from 'jsdom'
+
+const boostagrams_dir = 'src/boostagrams/custom_records'
+
+export function readBoostagrams() {
+  const boostagrams = fs
+    .readdirSync(boostagrams_dir)
+    .filter((paths) => paths.match(/custom_record_.*\.json/) !== null)
+    .map((path) => boostagrams_dir + '/' + path)
+    .map((path) => {
+      try {
+        return JSON.parse(fs.readFileSync(path))
+      } catch (err) {
+        console.log(`Couldn't parse ${path}. Skipping.`)
+        return null
+      }
+    })
+    .filter((json) => json !== null)
+
+  return filterAndFix(boostagrams)
+}
+
+function filterAndFix(boostagrams) {
+  return boostagrams
+    .filter((boostagram) => {
+      if (!boostagram.podcast && !boostagram.feedID && !boostagram.url) {
+        // Todo: Add `podcast:guid` once the podcast feed supports it.
+        return false
+      }
+
+      if (!!boostagram.podcast) return boostagram.podcast === 'Closing the Loop'
+      if (!!boostagram.feedID) return boostagram.feedID === '4058673'
+      if (!!boostagram.url) return boostagram.url === 'https://closing-the-loop.github.io/feed.xml'
+
+      if (!!boostagram.message) return boostagram.message.length > 0
+      else return false
+    })
+    .map((boostagram) => {
+      const maxLength = 280
+      var trimmed = boostagram.message.trim()
+      trimmed =
+        trimmed.length > maxLength ? trimmed.substring(0, maxLength - 3) + '...' : trimmed.substring(0, maxLength)
```

Comment:
```suggestion
        trimmed.length > maxLength ? trimmed.substring(0, maxLength - 3) + '…' : trimmed.substring(0, maxLength)
```

We can use the [ellipsis symbol](https://www.utf8icons.com/character/8230/horizontal-ellipsis) here, which is a single character: `…` 

I'm also not sure how JS `.substring()` handles multi-character stuff. There are some characters that actually consist of more than 1 character. Flag emojis, for example. So we might run into weird edge-cases. Just pointing it out because it's interesting, no reason to change anything here.

Reply:
Good suggestion! 💯

Reply:
Also, good point regarding character splitting thing! After a quick Google search it seems like the best thing to do is use a library for that e.g. [runes](https://github.com/dotcypress/runes). I'll add that real quick.

Reply:
Hm although maybe not that library. Doesn't seem to be maintained anymore...

Reply:
Fixed ✅